curriculum/challenges/english/07-scientific-computing-with-python/learn-recursion-by-solving-the-tower-of-hanoi-puzzle/64df45a3ad4f8719e5355244.md

```diff
@@ -16,7 +16,7 @@ For now, each recursive call prints the `rods` dictionary without performing any
 You should remove the last element from the `rods[source]` list and append it to the `rods[target]` list before the `print` call.
 
 ```js
-({ test: () => assert.match(code, /move\(\s*n\s*-\s*1\s*,\s*source\s*,\s*auxiliary\s*,\s*target\s*\)\s+rods\s*\[\s*target\s*\]\s*\.append\(\s*rods\s*\[\s*source\s*\]\s*\.pop\(\s*\)\s*\)/) })
+({ test: () => assert(runPython(`_Node(_code).find_function("move").find_body().find_if("n > 0").find_bodies()[0].find_calls("append")[0].is_equivalent("rods[target].append(rods[source].pop())")`)) })
```

Comment:
As far as checking if function moves things correctly at this point, something like this would work:
```suggestion
({ test: () => assert.isTrue(runPython(`
old_rods = rods
rods2 = {
    'A': list(range(5, 0, -1)),
    'B': [],
    'C': []
}
rods = rods2
print(rods)
move(5, 'A', 'B', 'C')
rods = old_rods
rods2['A'] == [] and rods2['C'] == list(range(1, 6))
`))})
```

But that doesn't check if the `print` is actually after the moves.

Reply:
You could use `is_ordered` here (it should work) https://opensource.freecodecamp.org/curriculum-helpers/python.html#is_ordered

Reply:
I have a question. Wouldn't checking the order of the statements potentially decrease the amount of available solutions? Like what if a camper uses a temporary variable to hold onto the popped value. I also don't know if checking to see the movement came before the `print` statement outweighs the cons.

Reply:
Hmm, what about, additionally, checking position of
```python    
    if n > 0:
        # move n - 1 disks from source to auxiliary, so they are out of the way
        move(n - 1, source, auxiliary, target)
```
and 
```python
        # display starting configuration
        print(rods, '\n')
```
If first one is at the start of the function, and second one is at the end, then added code, making function return correct result, must be in the middle.

Reply:
This makes sense. Unfortunately I am still struggling here and there manipulating the AST nodes.